include/silo_api/database_watcher.h

```diff
@@ -0,0 +1,25 @@
+#ifndef SILO_DATABASE_WATCHER_H
+#define SILO_DATABASE_WATCHER_H
+
+#include <string>
+
+#include <Poco/Delegate.h>
+#include <Poco/DirectoryWatcher.h>
+#include <Poco/Path.h>
+
+#include "silo_api/database_mutex.h"
+
+namespace silo_api {
+
+class DatabaseWatcher {
```

Comment:
Strictly speaking, this doesn't watch the database, it watches a directory where serialized databases might be stored. Rename to `DatabaseDirectoryWatcher` or `NewSerializedDatabaseStateWatcher` or similar?